### PR TITLE
PS-9739: implemented functionality to save totals mode

### PR DESCRIPTION
**Pull request by asmarovydlo**
Ticket: https://spryker.atlassian.net/browse/PS-9739
Epic: https://spryker.atlassian.net/browse/PS-8578

### CODE REVIEW

File: data/translation/Zed/de_DE.csv [hunk 1]
```diff
@@ -55,7 +55,7 @@ Default Business Unit,Standardgeschäftsbereich
 base64,base64
 url-encoded,url-encoded
 No Encoding,Keine Kodierung
-Request Mapping,Mapping anfragen
+Request Mapping,Mapping der Anfrage
```
Comment (odokienko) [inline data/translation/Zed/de_DE.csv:58]: Please check this translation.
```suggestion
Request Mapping,Mapping des Anfrage
```

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionCartForm.php [hunk 2]
```diff
@@ -175,4 +206,30 @@ function (string $maxDescriptionLength) {
...
+     */
+    protected function createTotalsModeValidationConstraint(): Constraint
+    {
+        return new Callback([
+            'callback' => function (string $totalsMode, ExecutionContextInterface $context) {
+                $connectionFormat = $context->getRoot()
+                     ->get(PunchoutCatalogConnectionTransfer::FORMAT)
+                    ->getData();
+
+                $allowedConnectionFormats = static::ALLOWED_CONNECTION_FORMATS_FOR_TOTALS_MODES[$totalsMode];
```
Comment (odokienko) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionCartForm.php]:
```suggestion
                $allowedConnectionFormats = static::ALLOWED_CONNECTION_FORMATS_FOR_TOTALS_MODES[$totalsMode] ?? [];
```

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionCartForm.php [hunk 3]
```diff
@@ -175,4 +206,30 @@ function (string $maxDescriptionLength) {
...
+
+    /**
+     * @return \Symfony\Component\Validator\Constraint
+     */
+    protected function createTotalsModeValidationConstraint(): Constraint
+    {
+        return new Callback([
+            'callback' => function (string $totalsMode, ExecutionContextInterface $context) {
+                $connectionFormat = $context->getRoot()
+                     ->get(PunchoutCatalogConnectionTransfer::FORMAT)
```
Comment (odokienko) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionCartForm.php:218]:
```suggestion
                    ->get(PunchoutCatalogConnectionTransfer::FORMAT)
```

**Review by odokienko — CHANGES_REQUESTED** [review]

File: data/translation/Zed/de_DE.csv (hunk 1 above)
Comment (asmarovydlo) [reply data/translation/Zed/de_DE.csv:58]: Done according to https://spryker.atlassian.net/browse/PS-9734?focusedCommentId=76638&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-76638

**Review by dmiseev — APPROVED** [review]

**Review by odokienko — APPROVED** [review]

**Review by demkos — APPROVED** [review]

**Pull request merged**